### PR TITLE
Add parameters to compute launch call

### LINKED ISSUE #156: Add all parameters exposed by compute API to the user
**What would you like to be added**:
Some missing items are

- CapacityReservationId
- DedicatedVmHostId
- Add tags to createvnicdetails
- HostnameLabel in createvnicdetails
- SkipSourceDestCheck in createvnicdetails
- LaunchOptions/BootVolumeType
- LaunchOptions/Firmware
- LaunchOptions/NetworkType
- LaunchOptions/RemoteDataVolumeType
- LaunchOptions/IsConsistentVolumeNamingEnabled
- InstanceOptions/AreLegacyImdsEndpointsDisabled
- AvailabilityConfig/IsLiveMigrationPreferred
- AvailabilityConfig/RecoveryAction
- PreemptibleInstanceConfig/PreemptionAction
- AgentConfig/IsMonitoringDisabled
- /AgentConfig/IsManagementDisabled
- AgentConfig/AreAllPluginsDisabled
- /AgentConfigPluginsConfig
- IsPvEncryptionInTransitEnabled
- InstanceSourceViaImageDetails/KmsKeyId
- InstanceSourceViaImageDetails/BootVolumeVpusPerGB
**Why is this needed**:

### PR DESCRIPTION
**Pull request by shyamradhakrishnan**
**What this PR does / why we need it**:

Add all launch parameters exposed by compute API(except for extendedMetadata)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #156

### CODE REVIEW

**Comment by shyamradhakrishnan** [comment]
unit tests
```
vtest_env /Users/shyaradh/oke/src/github.com/oracle/cluster-api-provider-oci/testbin; go test ./... -coverprofile cover.out && go tool cover -html=cover.out -o cover.html
Using cached envtest tools from /Users/shyaradh/oke/src/github.com/oracle/cluster-api-provider-oci/testbin
setting up env vars
?   	github.com/oracle/cluster-api-provider-oci	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/api/v1beta1	1.211s	coverage: 17.9% of statements
?   	github.com/oracle/cluster-api-provider-oci/cloud/config	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/cloud/ociutil	0.516s	coverage: 25.0% of statements
ok  	github.com/oracle/cluster-api-provider-oci/cloud/scope	167.121s	coverage: 73.2% of statements
?   	github.com/oracle/cluster-api-provider-oci/cloud/scope/mocks	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/base	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/base/mock_base	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/compute	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/compute/mock_compute	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/computemanagement	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/containerengine	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/containerengine/mock_containerengine	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/identity	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/identity/mock_identity	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer/mock_nlb	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/vcn	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/vcn/mock_vcn	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/controllers	28.270s	coverage: 69.0% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/api/v1beta1	1.954s	coverage: 10.1% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/controllers	1.884s	coverage: 43.4% of statements
?   	github.com/oracle/cluster-api-provider-oci/feature	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/version	[no test files]
```

**Comment by shyamradhakrishnan** [comment]
e2e test
```
Workload cluster creation
/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/test/e2e/cluster_test.go:50
  ClusterClass - with 1 control-plane nodes and 1 worker nodes [PRBlocking]
  /home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/test/e2e/cluster_test.go:471
------------------------------
STEP: Tearing down the management cluster


Ran 5 of 22 Specs in 1808.906 seconds
SUCCESS! -- 5 Passed | 0 Failed | 0 Pending | 17 Skipped
```

File: cloud/scope/machine.go [hunk 1]
```diff
@@ -239,6 +252,8 @@ func (m *MachineScope) GetOrCreateMachine(ctx context.Context) (*core.Instance,
 		IsPvEncryptionInTransitEnabled: common.Bool(m.OCIMachine.Spec.IsPvEncryptionInTransitEnabled),
 		FreeformTags:                   tags,
 		DefinedTags:                    definedTags,
+		//		ExtendedMetadata:               m.OCIMachine.Spec.ExtendedMetadata,
```
Comment (joekr) [inline cloud/scope/machine.go:255]: Remove commented code?

**Review by joekr — APPROVED** [review]
I'm almost wondering if we should be consuming Computes Spec files to build this list of parameters. I'm not sure how we will maintain this long term, but for now looks good. 😁

File: cloud/scope/machine.go (hunk 1 above)
Comment (shyamradhakrishnan) [reply cloud/scope/machine.go:255]: yep, will do in next PT

**Pull request merged**